### PR TITLE
First zoom implementation

### PR DESCRIPTION
**Pull request by schuellerf**
I just wanted so see the pixels "more clear" as zooming the picture afterwards makes it soo blurry :)

### CODE REVIEW

**Comment by Rogier-5** [comment]
It sounds like you are using photo editing software to magnify maps.

There are several ways of magnifying an image. Photo editing software usually interpolates the missing pixels in some way, or runs some filter on the result. That makes the magnified photo look less jagged and coarse, but also less sharp (more blurry) for its resolution. That is not desirable of course for minetest maps. You might try to disable the interpolating and/or the filter. The result will look better.

If you use [imagemagick](http://www.imagemagick.org/) to [resize](http://www.imagemagick.org/Usage/resize/), you might use:
```
convert map.png -scale 400% map4.png
```
for instance, to zoom by a factor 4.

**Comment by HybridDog** [comment]
the pictures could be converted to compressed svg vector graphics

**Comment by Rogier-5** [comment]
> the pictures could be converted to compressed svg vector graphics

I haven't tried, but I expect that for larger maps, it could be quite inefficient, memory-wise and/or storage-wise and/or cpu-time-wise. As svg is a textual format, the map will have to be converted to some form of text format for it to be included in the svg document...

**Comment by HybridDog** [comment]
l think the map tiles should use interlacing (Adam 7) and become loaded parallel,
so an undetailed map appears and gradually it becomes more and more detailed

**Comment by Rogier-5** [comment]
> l think the map tiles should use interlacing (Adam 7) and become loaded parallel,
so an undetailed map appears and gradually it becomes more and more detailed

That would be a nice feature indeed. 

Note that both SVG and interlacing are entirely unrelated to this pull request. I suggest we stick to the topic.

**Comment by HybridDog** [comment]
sorry

**Comment by sfan5** [comment]
Please use tabs (correctly and consistenly) instead of spaces.

**Comment by sfan5** [comment]
:+1: just remember to document the zoom option in README.rst

File: README.rst [hunk 1]
```diff
@@ -37,41 +37,45 @@ Parameters
...
-    Use specific map backend, supported: sqlite3, leveldb, redis, `--backend leveldb`
+    Use specific map backend, supported: *sqlite3*, *leveldb*, *redis*, e.g. ``--backend leveldb``
 
 geometry:
-    Limit area to specific geometry, `--geometry -800:-800+1600+1600`
+    Limit area to specific geometry, e.g. ``--geometry -800:-800+1600+1600``
+
+zoom:
+    "Zoom" the image by using more than one pixel per block.  
+    Currently this only uses one color per block (even when zoomed) e.g. ``--zoom 4``
```
Comment (sfan5) [inline README.rst]: Minor thing:
The term for 1x1 cubes is `node`.

File: TileGenerator.cpp [hunk 2]
```diff
@@ -592,15 +600,14 @@ inline void TileGenerator::renderShading(int zPos)
 			// more thickness -> less visible shadows: t=0 -> 100% visible, t=255 -> 0% visible
 			if (m_drawAlpha)
 				d = d * ((0xFF - m_blockPixelAttributes.attribute(z, x).thickness) / 255.0);
-			int sourceColor = m_image->tpixels[imageY][getImageX(x)] & 0xffffff;
+			int sourceColor = m_image->tpixels[getImageY(imageY)][getImageX(x)] & 0xffffff;
```
Comment (sfan5) [inline TileGenerator.cpp]: inlining the getImageY() here is actually less efficient, keep the `imageY = getImageY(imageY);` above.

**Comment by sfan5** [comment]
15fae27b85ad7c7875220685f11e2f1d5ccc9c41

**Pull request closed without merging**